packages/edit-post/src/components/header/template-title/delete-template.js

```diff
@@ -42,53 +48,57 @@ export default function DeleteTemplate() {
 		templateTitle = template.title;
 	}
 
+	const onDelete = () => {
+		clearSelectedBlock();
+		setIsEditingTemplate( false );
+
+		editPost( {
+			template: '',
+		} );
+		const settings = getEditorSettings();
+		const newAvailableTemplates = pickBy(
+			settings.availableTemplates,
+			( _title, id ) => {
+				return id !== template.slug;
+			}
+		);
+		updateEditorSettings( {
+			...settings,
+			availableTemplates: newAvailableTemplates,
+		} );
+		deleteEntityRecord( 'postType', 'wp_template', template.id );
+	};
+
 	return (
 		<MenuGroup className="edit-post-template-top-area__second-menu-group">
-			<MenuItem
-				className="edit-post-template-top-area__delete-template-button"
-				isDestructive
-				variant="secondary"
-				aria-label={ __( 'Delete template' ) }
-				onClick={ () => {
-					if (
-						// eslint-disable-next-line no-alert
-						window.confirm(
-							sprintf(
-								/* translators: %s: template name */
-								__(
-									'Are you sure you want to delete the %s template? It may be used by other pages or posts.'
-								),
-								templateTitle
-							)
-						)
-					) {
-						clearSelectedBlock();
-						setIsEditingTemplate( false );
-
-						editPost( {
-							template: '',
-						} );
-						const settings = getEditorSettings();
-						const newAvailableTemplates = pickBy(
-							settings.availableTemplates,
-							( _title, id ) => {
-								return id !== template.slug;
-							}
-						);
-						updateEditorSettings( {
-							...settings,
-							availableTemplates: newAvailableTemplates,
-						} );
-						deleteEntityRecord(
-							'postType',
-							'wp_template',
-							template.id
-						);
-					}
-				} }
-			>
-				{ __( 'Delete template' ) }
-			</MenuItem>
+			<>
+				<MenuItem
+					className="edit-post-template-top-area__delete-template-button"
+					isDestructive
+					variant="secondary"
+					aria-label={ __( 'Delete template' ) }
+					onClick={ () => {
+						setShowConfirmDialog( true );
+					} }
```

Comment:
Small optimisation: you could look into memoizing this callback so that the function doesn't get recreated every time the `DeleteMenu` component renders (see [related comment](https://github.com/WordPress/gutenberg/pull/37491#discussion_r780804949))